File: app/src/main/java/dolphin/android/apps/CpblCalendar/OnQueryCallback.java
package dolphin.android.apps.CpblCalendar;

import java.util.ArrayList;

import dolphin.android.apps.CpblCalendar.provider.CpblCalendarHelper;
import dolphin.android.apps.CpblCalendar.provider.Game;

/**
 * Created by dolphin on 2015/02/07.
 * <p>
 * Query action callback
 */
public interface OnQueryCallback {
    void onQueryStart();

    void onQueryStateChange(String msg);

    void onQuerySuccess(CpblCalendarHelper helper, ArrayList<Game> gameArrayList, int year, int month);

    void onQueryError(int year, int month);
}

File: app/src/main/java/dolphin/android/apps/CpblCalendar/OnQueryCallback.kt
package dolphin.android.apps.CpblCalendar

import dolphin.android.apps.CpblCalendar.provider.CpblCalendarHelper
import dolphin.android.apps.CpblCalendar.provider.Game
import java.util.*

/**
 * Created by dolphin on 2015/02/07.
 *
 *
 * Query action callback
 */
interface OnQueryCallback {
    fun onQueryStart()

    fun onQueryStateChange(msg: String?)

    fun onQuerySuccess(helper: CpblCalendarHelper?, gameArrayList: ArrayList<Game>?, year: Int, month: Int)

    fun onQueryError(year: Int, month: Int)
}
